shared/test/test_files.rb

```diff
@@ -425,6 +425,64 @@ def test_rename_mixed_case
     delete_all_manifest_versions
   end
 
+  def test_rename_case_only
+    filename = @api.randomize_filename('Mixed Case With Spaces.png')
+    escaped_filename = URI.escape(filename)
+    filename2 = filename.sub 'Mixed Case', 'mixeD casE'
+    escaped_filename2 = URI.escape(filename2)
+    delete_all_file_versions filename, filename2
+    delete_all_manifest_versions
+
+    image_body = 'stub-image-contents'
+
+    post_file_data(@api, filename, image_body, 'image/png')
+    assert successful?
+
+    @api.get_object(escaped_filename)
+    assert successful?
+    assert_equal image_body, last_response.body
+
+    @api.get_object(escaped_filename2)
+    assert successful?
+    assert_equal image_body, last_response.body
+
+    get "v3/files/#{@channel_id}"
+    response_before_rename = JSON.parse(last_response.body)
+    # There should be only one file with filename, category, and size matching our expectations
+    expected_image_info = {'filename' => filename, 'category' => 'image', 'size' => image_body.length}
+    file_infos = response_before_rename['files']
+    assert_equal(1, file_infos.length)
+    assert_fileinfo_equal(expected_image_info, file_infos[0])
+
+    @api.rename_object(filename, escaped_filename2)
+    assert successful?
+
+    @api.get_object(escaped_filename)
+    assert successful?
+    assert_equal image_body, last_response.body
+
+    @api.get_object(escaped_filename2)
+    assert successful?
+    assert_equal image_body, last_response.body
+
+    get "v3/files/#{@channel_id}"
+    response_after_rename = JSON.parse(last_response.body)
+    # There should be only one file with the new filename, category, and size matching our expectations
+    expected_image_info_after_rename = {'filename' => filename2, 'category' => 'image', 'size' => image_body.length}
+    file_infos_after_rename = response_after_rename['files']
+    assert_equal(1, file_infos_after_rename.length)
+    assert_fileinfo_equal(expected_image_info_after_rename, file_infos_after_rename[0])
+
+    # The manifest version (filesVersionId) should be different, but the file version should be the same
+    assert response_before_rename['filesVersionId'] != response_after_rename['filesVersionId']
```

Comment:
Perhaps use `refute_equal` here, as doing so may provide a more helpful error message should the test fail here.